### DISCUSSION TITLE
Reset osqp workspace with codegen

### DISCUSSION THREAD

**Discussion by charleskhazoom** [osqp/osqp#579] [Embedded Optimization]
I am working with the codegen feature from Matlab and loading the generated c code in c++.

Is is possible to reset the osqp workspace between runs ? For instance, I am running two unit tests from the same executable that both include the same osqp workspace. As a result, it seems that the tests are coupled, which makes the testing difficult.

**Comment by imciner2** [comment]
Sorry for the delay.

The OSQP workspace is solver-specific, so if you create a new solver object and initialize it, you will get a new workspace.
